### PR TITLE
#162377074 Roles fields can check if user input is valid.

### PR DESCRIPTION
**Pull request by mikenthiwa**
#### What does this PR do?
 - Role field can check if the user input is in the list of roles.
 - Dropdown can automatically close by clicking outside the field.

#### Description of Task to be completed?
 - When the user fills in a role that is not in the list of roles, an error will be displayed.
 - The user can automatically close the drop-down by clicking outside the role field.  

#### How should this be manually tested?
 - Git clone this repository `https://github.com/andela/travel_tool_front.git`.
 - Checkout to` fix-input-error-162377074`.
 - Install dependancies `yarn install`.
 - Start app `yarn start`.
 - Go to `Profile page`.
 - Fill an invalid role and submit.
 - Repeat the same for request page.

#### Screenshots (if appropriate)
screen shot 2018-12-07 at 14 32 49

screen shot 2018-12-07 at 14 46 26

#### What are the relevant pivotal tracker stories?
[162377074](https://www.pivotaltracker.com/story/show/162377074)

### CODE REVIEW

File: src/components/Forms/NewAccommodationForm/NewAccommodation.jsx [hunk 1]
```diff
@@ -1,3 +1,4 @@
+/* eslint-disable */
```
Comment (ssewilliam) [inline src/components/Forms/NewAccommodationForm/NewAccommodation.jsx]: Why are we disabling ES Lint here

File: src/redux/middleware/UserProfileSaga.js [hunk 2]
```diff
@@ -14,10 +14,14 @@ export function* postUserProfileDataSagaAsync(action) {
     if (action.showToast){
       toast.success('Profile updated successfully');
     }
-  } catch (error) { /* istanbul ignore next */
+  } catch (error) {
     const errorMessage = apiErrorHandler(error);
-    /* istanbul ignore next */
-    toast.error(errorMessage);
+    if (errorMessage === 'occupation is required'){ /* istanbul ignore next */
+      toast.error('the role you entered does not exist');
```
Comment (ssewilliam) [inline src/redux/middleware/UserProfileSaga.js]: I think this line can also start with **T**

**Comment by mikenthiwa** [comment]
@samuelkubai Yes, I can sort that one out.

**Comment by samuelkubai** [comment]
@mikenthiwa are you working on the feedback provided?

**Comment by Spectrumsun** [comment]
@mikenthiwa we still have a bug on the form

**Comment by Spectrumsun** [comment]
works well @mikenthiwa  fix failing test

**Pull request merged**